ci/test_python.sh

```diff
@@ -12,7 +12,7 @@ rapids-logger "Create test conda environment"
 rapids-dependency-file-generator \
   --output conda \
   --file-key test_python \
-  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};dependencies=${RAPIDS_DEPENDENCIES}" | tee env.yaml
```

Comment:
For wheel tests, we will have to engineer a way to use the oldest dependencies. We determine the test dependencies at _build_ time, via `[test]` extras. However, for this case, we really just want to add some extra pinnings at test time based on the value of `RAPIDS_DEPENDENCIES`. We may have to add a call in `ci/test_wheel.sh` that invokes `rapids-dependency-file-generator` to create a `requirements.txt` file that we install after installing the previously-built wheel.

Reply:
Nice, that makes sense to me (and it is really cool to see how straight forward this is looking like).
Don't want to mess with the builds, just testing, so branching between using `package[test]` and a custom requirements install on `RAPIDS_DEPENDENCIES="earliest"` seems good.